packages/integrations/mdx/src/index.ts

```diff
@@ -149,14 +185,20 @@ export default function mdx(partialMdxOptions: Partial<MdxOptions> = {}): AstroI
 	};
 }
 
-const defaultOptions: MdxOptions = {
-	...markdownConfigDefaults,
+const defaultMdxOptions = {
 	extendMarkdownConfig: true,
 	recmaPlugins: [],
-	remarkPlugins: [],
-	rehypePlugins: [],
-	remarkRehype: {},
-};
+}
+
+function markdownConfigToMdxOptions(markdownConfig: typeof markdownConfigDefaults): MdxOptions {
```

Comment:
Removing the `any` type per @ematipico's suggestion uncovered a type error! This refactor ensures all Markdown config options are properly converted _before_ getting passed down to the MDX compiler